src/bench/node_allocator.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2019-2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bench/bench.h>
+#include <coins.h>
+#include <memusage.h>
+#include <support/allocators/node_allocator.h>
+
+#include <cstring>
+#include <iostream>
+#include <unordered_map>
+
+template <typename Map>
+void BenchFillClearMap(benchmark::Bench& bench, Map& map)
+{
+    CMutableTransaction tx = CMutableTransaction();
+    tx.vin.resize(1);
+    tx.vin[0].scriptSig = CScript() << OP_2;
+    tx.vin[0].scriptWitness.stack.push_back({2});
+    tx.vout.resize(1);
+    tx.vout[0].scriptPubKey = CScript() << OP_2 << OP_EQUAL;
+    tx.vout[0].nValue = 10 * COIN;
+
+    COutPoint p{tx.GetHash(), 0};
+
+    bench.epochIterations(5000 * 10).run([&] {
+        // modify hash a bit so we get a new entry in the map
+        ++p.n;
```

Comment:
I think `p.n` cannot overflow here. It is initialized to 0, and since I've specified the number of iterations each measurement will have exactly 50000 iterations. I do this so I can be sure each measurement has exactly the same number of calls to `map.clear();`, which gives more stable benchmark results. Nanobench defaults to 11 measurements, so in total n will reach 550000.